Framework/src/TaskRunner.cxx

```diff
@@ -59,7 +63,40 @@ TaskRunner::TaskRunner(const TaskRunnerConfig& config)
 {
 }
 
-void TaskRunner::init(InitContext& iCtx)
+void TaskRunner::refreshConfig(InitContext& iCtx)
+{
+  try {
+    // get the tree
+    auto updatedTree = iCtx.options().get<boost::property_tree::ptree>("qcConfiguration");
+
+    if (updatedTree.empty()) {
+      ILOG(Warning, Devel) << "Templated config tree is empty, we continue with the original one" << ENDM;
+    } else {
+      if (gSystem->Getenv("O2_QC_DEBUG_CONFIG_TREE")) { // until we are sure it works, keep a backdoor
+        ILOG(Debug, Devel) << "We print the tree we got from the ECS via DPL : " << ENDM;
+        printTree(updatedTree);
+      }
+
+      // prepare the information we need
+      auto infrastructureSpec = InfrastructureSpecReader::readInfrastructureSpec(updatedTree);
+      // find the correct taskSpec
+      auto it = find_if(infrastructureSpec.tasks.begin(),
+                        infrastructureSpec.tasks.end(),
+                        [this](const TaskSpec& ts) { return ts.taskName == mTaskConfig.taskName; });
+      if (it != infrastructureSpec.tasks.end()) {
+        mTaskConfig = TaskRunnerFactory::extractConfig(infrastructureSpec.common, *it, mTaskConfig.parallelTaskID, it->resetAfterCycles);
```

Comment:
`it->resetAfterCycles`
That will break if we have a multinode setup with local tasks and delta mergers. See:
https://github.com/AliceO2Group/QualityControl/blob/95eb565ef04146919c2034099e26289ae766230e/Framework/src/InfrastructureGenerator.cxx#L136

It seems to me that maybe we should somehow extract such logic and use the same function in `InfrastructureGenerator` and here?

Reply:
Good that you spotted that, I completely overlooked the issue.